packages/utils/src/internals/sitemap.ts

```diff
@@ -18,6 +20,42 @@ class ParsingState {
     }
 }
 
+class SitemapTxtParser extends Writable {
+    private decoder: StringDecoder = new StringDecoder('utf8');
+    private buffer: string = '';
+
+    constructor(private parsingState: ParsingState, private onEnd: () => void) {
+        super();
+    }
+
+    private processBuffer(input: string, finalize: boolean): void {
+        this.buffer += input;
+
+        if (finalize || this.buffer.includes('\n')) {
+            const parts = this.buffer.split('\n').filter((part) => part.length > 0);
+
+            if (finalize) {
+                this.parsingState.urls.push(...parts);
+                this.buffer = '';
+            } else if (parts.length > 0) {
+                this.parsingState.urls.push(...parts.slice(0, -1));
+                this.buffer = parts.at(-1)!;
+            }
+        }
+    }
```

Comment:
Splitting on `\n` leaves in `\r` at the end of each URL in case the sitemap.txt file is `CRLF` formatted. This + the way Node.JS stringifies `\r` results in invalid URLs:
![obrazek](https://github.com/apify/crawlee/assets/61918049/e2488faa-4bda-4eb3-8154-678ea222ac1d)

Maybe it's safer to `replaceAll('\r', '')` after the URLs are split?

Reply:
I guess a `.trim` would work even better, am I right?